### ISSUE TITLE
Bug: callpeak --broad uses 4*tagsize as lvl2_max_gap

### ISSUE THREAD

**Issue by knutdrand** [taoliu/MACS#365]
**Describe the bug**
From 97e27a0dc685123ef5d89a8ed5601423b0b724a6 it seems callpeak --broad uses 4\*tagsize when merging level 2 peaks, instead of 4\*fragmentsize which was used earlier.  
I cannot find this change described in the change log, and the readme still says 
"The maximum length of broad region length is 4 times of d from MACS."

**To Reproduce**
L253 of PeakDetect.pyx

**Expected behavior**
Either 4*fragmentsize should be used, or readme and changelog updated.

**Comment by taoliu** [comment]
Thanks for the feedback! I definitely need to update that line in the README. 

As for the default value, `tagsize` is for merging nearby smalls peaks for level 1 peak (strong peak), and the corresponding option is `--max-gap`. Then 4 times of this value is for merging level 2 peak (weak peak). The minimal length of peaks (no matter level 1 or level 2 in broad peak setting) is always `fragment size`.

**Comment by knutdrand** [comment]
Thanks for the quick reply!
Am I then correct in that if I set `--max-gap` to the estimated fragment length, I should get the same `.broadPeak` as with default parameters before v2.1.2, but not necessarily the same `.gappedPeak`, due to different merging of level 1 peaks?

And on a side note, do you think the new default parameters are in general more sensible than the old ones? Was 4*fragment_length too lenient?

**Comment by taoliu** [comment]
@knutdrand Sorry for the late reply! My reply about the default values of `max-gap` should also apply to v2.1.2. In other words,  MACS 2.1.2 by default uses 4*`tagsize` as `max-gap` to merge nearby broad regions (lvl2 peaks), and `tagsize` to merge nearby lvl peaks. The description of 4*`fragment_length` in README is wrong. I have to update that line in the README. If you read the message from command line `macs2 callpeak -h` and you will see the correct description:

```
  --broad               If set, MACS will try to call broad peaks using the
                        --broad-cutoff setting. Please tweak '--broad-cutoff'
                        setting to control the peak calling behavior. At the
                        meantime, either -q or -p cutoff will be used to
                        define regions with 'stronger enrichment' inside of
                        broad peaks. The maximum gap is expanded to 4 * MAXGAP
                        (--max-gap parameter). As a result, MACS will output a
                        'gappedPeak' and a 'broadPeak' file instead of
                        'narrowPeak' file. Note, a broad peak will be reported
                        even if there is no 'stronger enrichment' inside.
                        DEFAULT: False
```

And this is the same from 2.1.2 to the most recent 2.2.6. Sorry for the confusion!

**Comment by knutdrand** [comment]
Hi, sorry I was a bit unclear. 
I mean that in 2.1.1 for instance, the broad peaks are merged with `lvl2_max_gap=self.d*4`. While from 2.1.2 and onwards they are merged with `lvl2_max_gap=self.maxgap*4`.
But `self.d` is the fragmentlength, while the default value for `maxgap` is the tag length. 
So the broad peaks called with v2.1.1 is quite different from the peaks called with 2.1.2 and onwards (when using default parameters).

**Comment by taoliu** [comment]
@knutdrand Oh, sorry that I misunderstood your question! If we are discussing v2.1.1, I have to rewind my mind to 4 years ago :)  Yes. I checked the source code of 2.1.1, and the 4*`self.d` was used to merge lvl2 peaks.  It is indeed too lenient as you said.

**Comment by knutdrand** [comment]
Thanks, that clears it up!

**Comment by taoliu** [comment]
You are welcome! I have fixed the description in PR#380 and in the v2.2.7.1 release.
